java/maven/src/org/netbeans/modules/maven/NbMavenProjectImpl.java

```diff
@@ -639,6 +640,24 @@ public MavenEmbedder getEmbedder() {
         return uris.toArray(new URI[uris.size()]);
     }
 
+    private static File findJavaRoot(File root, boolean test) {
+        String javaFolder = test ? "test" + File.separator + "java" : "main" + File.separator + "java"; // NOI18N
+        File newRoot = new File(root, javaFolder);
+        if (newRoot.isDirectory()) {
+            return newRoot;
+        }
```

Comment:
This will recognize `target/generated-sources/test/java` and `target/generated-sources/test/java` - did someone seriously place files there?